kratos/containers/pointer_hash_map_set.h

```diff
@@ -64,7 +64,7 @@ template<class TDataType,
          class THashType = std::hash<TDataType>,
          class TGetKeyType = SetIdentityFunction<TDataType>,
          class TPointerType = Kratos::shared_ptr<TDataType> >
-class PointerHashMapSet
```

Comment:
@pooyan-dadvand we need your comment on the changes in the base containers (which make sense to us).

specifically i am ok with the changes to csr related stuff